Cubical/Data/FinWeak/Base.agda

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS --safe #-}
+module Cubical.Data.FinWeak.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+
+open import Cubical.Data.Bool
+open import Cubical.Data.Nat using (ℕ; zero; suc)
+
+open import Cubical.Relation.Nullary
+
+private variable
+  n : ℕ
+
+data FinPure : ℕ → Type where
+  zero : FinPure 1
+  suc  : FinPure n → FinPure (suc n)
+
+data Fin : ℕ → Type where
+  pure   : FinPure n → Fin n
+  weaken : Fin n     → Fin (suc n)
+
+-- useful patterns
```

Comment:
I think this comment can be removed, since I would assume that only useful patterns are introduced anyway.